src/aff_expr.jl

```diff
@@ -337,7 +337,7 @@ function MOI.VectorAffineFunction(affs::Vector{AffExpr})
     MOI.VectorAffineFunction(terms, constant)
 end
 moi_function(a::Vector{<:GenericAffExpr}) = MOI.VectorAffineFunction(a)
-function moi_function_type(::Type{Vector{Aff}}) where {T, Aff <: GenericAffExpr{T}}
+function moi_function_type(::Type{<:Vector{<:GenericAffExpr{T}}}) where {T}
```

Comment:
Why is the `<:` needed before Vector?

Reply:
Consider this example:
```julia
julia> f(::Type{Vector{<:Any}}) = 1
f (generic function with 1 method)

julia> f(Vector{Int})
ERROR: MethodError: no method matching f(::Type{Array{Int64,1}})
Closest candidates are:
  f(::Type{Array{T,1} where T}) at REPL[1]:1
Stacktrace:
 [1] top-level scope at none:0

julia> f(::Type{<:Vector{<:Any}}) = 1
f (generic function with 2 methods)

julia> f(Vector{Int})
1
```
The type `Vector{Int}` is the the same as the type `Vector{<:Any}` but it is a subtype of `Vector{<:Any}`.

Reply:
What was wrong  with
```julia
::Type{Vector{S}}) where {T, S <: GenericAffExpr{T}}
```
It is easier to understand than the two `<:`

(Although I now see this is used a lot.)

Reply:
I find it confusing to create a type variable that is not used when we can avoid it. I can revert it if you don't share this feeling

Reply:
I'm not fussy. We probably need a style point on type signatures.